cognite/client/data_classes/iam.py

```diff
@@ -51,24 +54,58 @@ def _load(cls, resource: dict, cognite_client: CogniteClient | None = None) -> S
         return cls(
             name=resource["name"],
             source_id=resource.get("sourceId"),
-            capabilities=[Capability.load(c) for c in resource.get("capabilities", [])],
-            id=resource.get("id"),
-            is_deleted=resource.get("isDeleted"),
+            capabilities=[Capability.load(c) for c in resource.get("capabilities", [])] or None,
+            id=resource["id"],
+            is_deleted=resource["isDeleted"],
```

Comment:
This class is used for both read and write, and `id` and `isDeleted` are not required for the `write` version. So I think they need to be optional.

Reply:
Oh right... yeah guess so; will revert. Would be awkward to force people to define it as `id: None, isDeleted: None` in the config :(